container-search/src/main/java/com/yahoo/prelude/query/SameElementItem.java

```diff
@@ -58,8 +58,8 @@ protected void adding(Item item) {
 
     @Override
     public Optional<Item> extractSingleChild() {
-        if (getItemCount() == 1) {
-            WordItem child = (WordItem) getItem(0);
+        if (getItemCount() == 1 && getItem(0) instanceof SimpleIndexedItem) {
```

Comment:
No need to check for SimpleIndexedItem here as that is required to be a TermItem insertion time.

Reply:
I am not sure why it was limited to TermItem initially. I remember we had some discussions about it. I will redo them and see if we can relax requirement.

Reply:
Yes but if that ever changes we have a new latent bug ... 
It's close in the code though, so I don't feel strongly about it.

Reply:
If so we will have a bug we will be able to spot, instead of a silent one we will never see..
There are tests that inserts fail with wrong Item so I think keeping the error checking in one place is a lot cleaner.

Reply:
We'd only have a bug if sameElement with one child doesn't work in the backend. Anyway, fine  :-)

Reply:
You will get a silent performance degradation.... Which was the reason for introducing the extractSingleChild :)

I checked with the backend and saw that only TermItems were handled. So before allowing anything else I need to verify with @havardpe if could allow at least weghted set.